pycloudlib/ibm_softlayer/instance.py

```diff
@@ -0,0 +1,374 @@
+            self._instance = self._vs_manager.get_instance(self.id)
+        # if IP address is still not available, raise exception
+        if "primaryIpAddress" not in self._instance:
+            raise IBMSoftlayerException(
+                f"Failed to get IP address for instance {self.id}"
+            )
+
+        return self._instance["primaryIpAddress"]
+
+    @staticmethod
+    def create_raw_instance(
+        vs_manager: SoftLayer.VSManager,
+        target_image_global_identifier: str,
+        hostname: str,
+        flavor: str,
+        datacenter: str,
+        public_security_group_ids: List[int],
+        private_security_group_ids: List[int],
+        ssh_key_ids: List[int],
+        domain_name: str,
+        **kwargs,
+    ):
+        """
+        Verify instance configuration and create instance.
+
+        Args:
+            vs_manager: Softlayer VSManager (Virtual Server Manager) instance
+            target_image_global_identifier: image global identifier
+            hostname: instance hostname
+            flavor: instance flavor
+            datacenter: datacenter region
+            public_security_group_ids: list of public security group ids
+            private_security_group_ids: list of private security group ids
+            ssh_key_ids: list of ssh key ids
+            domain_name: domain name
+
+        """
+        logger.debug("Creating raw instance")
+        constant_args = {
+            "private": False,
+            "dedicated": False,  # default
+            "hourly": True,  # default
+            "local_disk": False,  # default
+        }
+        instance_specific_args = {
+            "domain": domain_name,
+            "hostname": hostname,
+            "ssh_keys": ssh_key_ids,
+            "image_id": target_image_global_identifier,
+            "flavor": flavor,
+            "datacenter": datacenter,
+            "public_security_groups": public_security_group_ids,
+            "private_security_groups": private_security_group_ids,
+        }
+        # check if instance configuration is valid
+        try:
+            logger.info(
+                "Verifying configuration for instance before creating it."
+            )
+            vs_manager.verify_create_instance(
+                **constant_args,
+                **instance_specific_args,
+                **kwargs,
+            )
+        except SoftLayer.SoftLayerAPIError as e:
+            logger.error("configuration for instance is invalid: %s", e)
+            raise IBMSoftlayerException(
+                f"Failed to verify instance configuration: {e}"
+            ) from e
+        except Exception as e:
+            logger.error(
+                "Unexpected error while verifying instance configuration: %s",
+                e,
+            )
+            raise IBMSoftlayerException(
+                f"Unexpected error while verifying instance configuration: {e}"
+            ) from e
+        else:
```

Comment:
Nit: As every except clause raises, could we delete the else and deindent the last block of code?

Reply:
haha good nit! will do